Sources/XiEditor/LineCache.swift

```diff
@@ -60,6 +54,32 @@ struct Line<T> {
     }
 }
 
+enum AnnotationType: String, CaseIterable {
+    case Selection = "selection"
+    case Find = "find"
+}
+
+/// Represents an annotation (eg. selection, find highlight)
+struct Annotation {
+    let startLine: Int
```

Comment:
we do have a `BufferPosition` type defined in `EditView.swift`, that might be useful here? Although also fine as-is.